src/main/java/net/javadiscord/javabot/systems/help/HelpListener.java

```diff
@@ -54,40 +48,58 @@ public class HelpListener extends ListenerAdapter implements ButtonHandler {
 	private final Map<Long, Long> recentlyCloseSuggestedPosts = new LinkedHashMap<>(8, 0.75f, true) {
 		@Override
 		protected boolean removeEldestEntry(Map.Entry<Long, Long> eldest) {
 			return System.currentTimeMillis() > eldest.getValue() || size() >= 32;
 		}
 	};
-	private final UserPreferenceService preferenceService;
+	private final AutoCodeFormatter autoCodeFormatter;
+
+	/**
+	 * The constructor of this class, which is set corresponding to lombok.
+	 *
+	 * @param autoMod                   automod reference.
+	 * @param preferenceService         UserPreferenceService reference
+	 * @param botConfig                 BotConfig reference
+	 * @param helpTransactionRepository HelpAccountRepository reference
+	 * @param helpAccountRepository     HelpAccountRepository reference
+	 * @param dbActions                 DbActions reference
+	 * @param experienceService         ExperienceService reference
+	 */
+	public HelpListener(AutoMod autoMod, UserPreferenceService preferenceService, BotConfig botConfig, HelpAccountRepository helpAccountRepository, HelpTransactionRepository helpTransactionRepository, HelpExperienceService experienceService, DbActions dbActions) {
+		this.preferenceService = preferenceService;
+		this.botConfig = botConfig;
+		this.helpAccountRepository = helpAccountRepository;
+		this.helpTransactionRepository = helpTransactionRepository;
+		this.experienceService = experienceService;
+		this.dbActions = dbActions;
+		this.autoCodeFormatter = new AutoCodeFormatter(autoMod, botConfig, preferenceService);
+	}
 
 	@Override
 	public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+		handleMessageEvent(event);
+		autoCodeFormatter.handleMessageEvent(event);
+	}
+
+	private void handleMessageEvent(@NotNull MessageReceivedEvent event) {
 		// listen for the posts initial message. Why this has to be done is further described in the JDA Discord:
 		// https://canary.discord.com/channels/125227483518861312/1053705384466059354/1053705384466059354
 		if (newThreadChannels.contains(event.getChannel().getIdLong())) {
 			HelpConfig config = botConfig.get(event.getGuild()).getHelpConfig();
 			ThreadChannel post = event.getChannel().asThreadChannel();
 			// send post buttons
-			post.sendMessageComponents(ActionRow.of(
-					createCloseSuggestionButton(post),
-					Button.secondary(ComponentIdBuilder.build(HelpManager.HELP_GUIDELINES_IDENTIFIER), "View Help Guidelines")
-			)).queue(success -> post.sendMessageFormat(config.getReservedChannelMessageTemplate(), UserSnowflake.fromId(post.getOwnerId()).getAsMention(), config.getInactivityTimeoutMinutes()).queue());
+			post.sendMessageComponents(ActionRow.of(createCloseSuggestionButton(post), Button.secondary(ComponentIdBuilder.build(HelpManager.HELP_GUIDELINES_IDENTIFIER), "View Help Guidelines"))).submit();
+			post.sendMessageFormat(config.getReservedChannelMessageTemplate(), UserSnowflake.fromId(post.getOwnerId()).getAsMention(), config.getInactivityTimeoutMinutes()).submit();
```

Comment:
Why did you split that up into two calls happening at the same time? The second call was sent after the first one in order to ensure the message is sent after the first message.
However, it might be a good idea to combine both into a single message.